internal/pkg/grpc/client.go

```diff
@@ -122,6 +122,11 @@ func (c *Client) SendRequest(serviceMethod string, message string, headers []str
 		interpolatedHeaders[i] = placeholders.InterpolatePlaceholders(header)
 	}
 
+	if c.conn == nil {
+		log.Printf("No connection available. Skip making request.")
+		return response.Response{Duration: time.Duration(0), Err: err, Type: respType}
```

Comment:
The error we return here is defined in a different scope, when calling `RequestParserAndFormatter`.
Can we create a new error for this scope and return it instead?

Reply:
Thanks, I updated the pr.